aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1645,9 +1645,11 @@ Tensor sparse_compressed_to_sparse_bsr(const Tensor& self, IntArrayRef blocksize
     }
     return _compressed_to_block_compressed_cpu<kSparseBsr>(self.cpu(), blocksize).to(self.device());
   }
-  AT_ERROR(
-      "sparse_compressed_to_sparse_bsr expected SparseCsr, SparseBsr or SparseBsc layout but got ",
-      self.layout());
+  if (self.layout() == kSparseCsc) {
+    return self.to_sparse_csr(dense_dim_opt).to_sparse_bsr(blocksize);
+  }
```

Comment:
We have the COO -> BSR conversion. The path `self.to_sparse().to_sparse_bsr()` should be faster and it also runs on `self.device`.

Reply:
Well, let's at least follow this fast path for non-batched inputs?

Reply:
As mentioned above, existing code for these conversions ({CSR,CSC}->{BSR,BSC}) does not support batched inputs anyway - so it would mean throwing away this code.  I agree with your suggestion, and would to that; anyone else willing to cast a vote?

Reply:
Sorry, @alexsamardzic , I missed your point regarding batch dimensions. Still leaving just `to_sparse().to_block_spares()` is not correct since there is no notion of batch dimensions for COO inputs.

Reply:
My bad, I realized later that COO->BSR and COO->BSC conversions are actually implemented as COO->CSR->BSR and COO->CSC->BSC, respectively; so going through COO for CSR->BSR and CSC->BSC conversions is not possible at the moment.

Adding support for batch dimensions for CSR->BSR and CSC->BSC is not just about an additional outer loop, as it could for example happen that the number of NNZ elements is not the same across batch dimensions.  It's certainly doable, but I'd prefer to have this PR merged as is, and support for batch dimensions eventually added as separate PR.

Reply:
Hm, I see, indeed. We could revisit this part later then.